src/Mimi.jl

```diff
@@ -269,6 +267,11 @@ function addparameter(m::Model, name::Symbol, value)
             p = UncertainArrayParameter(value)
             m.parameters[symbol(lowercase(string(name)))] = p
         else
+            if !(typeof(value) <: Array{m.numberType})
+                # E.g., if model takes Number and given Float64, convert it
+                value = convert(Array{m.numberType}, value)
+            end
```

Comment:
This is good for now, but hopefully we won't need it anymore once we generate the types just before the ``run`` call because then we can just make the type of the parameter match. But given that that is all future talk, this is really super helpful now.